packages/flame/lib/src/extensions/vector2.dart

```diff
@@ -75,6 +75,15 @@ extension Vector2Extension on Vector2 {
     }
   }
 
+  /// Ensure that the vector is no larger than [max] in magnitude. Scales the
+  /// vector down without changing its direction if necessary.
+  void clampMagnitude(double max) {
```

Comment:
Is this needed when `clampScalar` exists in the original class?

Reply:
`clampScalar` will constrain the vector to lie in the box `[min, max] x [min, max]`. So this does not ensure maximum length of the vector, and will also cause the vector to change the direction.

But perhaps `clampLength` might be a better name here.

Reply:
https://github.com/flame-engine/flame/blob/main/packages/flame/lib/src/extensions/vector2.dart#L71

Reply:
`scaleTo` scales the vector to the desired length, this function scales the vector only if its length already exceeds `max`. So, it is equivalent to `if (vector.length > max) { vector.scaleTo(max); }`, except that it avoids computing the length of the vector twice.